File: 2021/test/day_06_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_06"

class Day06Test < Minitest::Test
  def test_part1
    day = Day06.new

    assert_equal 5934, day.part1(input("day_06_example_1"))
    assert_equal 359_344, day.part1(input("day_06"))
  end

  def test_part2
    day = Day06.new

    assert_equal 26_984_457_539, day.part2(input("day_06_example_1"))
    assert_equal 1_629_570_219_571, day.part2(input("day_06"))
  end
end

File: 2025/test/day_06_test.exs
defmodule Day06Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day06.part_1(input("day_06_example_1", false)) == 4_277_556
    assert Day06.part_1(input("day_06", false)) == 6_635_273_135_233
  end

  test "part2" do
    assert Day06.part_2(input("day_06_example_1", false)) == 3_263_827
    assert Day06.part_2(input("day_06", false)) == 12_542_543_681_221
  end
end
